src/coreComponents/linearAlgebra/utilities/LinearSolverParameters.hpp

```diff
@@ -66,10 +66,24 @@ struct LinearSolverParameters
   integer logLevel = 0;     ///< Output level [0=none, 1=basic, 2=everything]
   integer dofsPerNode = 1;  ///< Dofs per node (or support location) for non-scalar problems
   bool isSymmetric = false; ///< Whether input matrix is symmetric (may affect choice of scheme)
+  integer stopIfError = 1;  ///< Whether to stop the simulation if the linear solver reports an error
 
   SolverType solverType = SolverType::direct;                        ///< Solver type
   PreconditionerType preconditionerType = PreconditionerType::iluk;  ///< Preconditioner type
 
+  /// Direct solver parameters: used for SuperLU_Dist interface through hypre and PETSc
+  struct Direct
+  {
+    real64 relTolerance = 1.e-12;   ///< Tolerance used to check a direct solver solution
+    integer equilibrate = 1;        ///< Whether to scale the rows and columns of the matrix
+    string colPerm = "metis";       ///< How to permute the columns [none, MMD_At+A, MMD_AtA, colAMD,
+                                    ///< metis, parmetis]
+    string rowPerm = "mc64";        ///< How to permute the rows [none, mc64]
```

Comment:
So... with the latest `develop`, these can now become enumerations instead of strings. If you'd like to do this in current PR, the steps are roughly (similar to `solverType` and `preconditionerType`):
1. Declare the enum (e.g. `ColPermType`) inside `struct Direct` and change the type of `colPerm` variable
1. `#include <common/EnumStrings.hpp>` (already done)
1. `ENUM_STRINGS( LinearSolverParameters::Direct::ColPermType, "none", "MMD_At+A", "colAMD", "metis", "parmetis" )` after the closing brace of `LinearSolverParameters`
1. Where this variable is registered with a wrapper, you can get the list of options to put in description via `EnumStrings< LinearSolverParameters::Direct::ColPermType >::concat()`, and remove the validation of input since it will now be automatic
1. Where this variable is used, trivially change the conversion functions into SuperLUDist options to accept the new type instead of `string`
1. Same steps for `rowPerm`

Reply:
Let's address this in the current PR, which will in any case follow #1123.

Reply:
Done!